llvm/lib/Support/Timer.cpp

```diff
@@ -380,8 +381,13 @@ void TimerGroup::PrintQueuedTimers(raw_ostream &OS) {
 
   // Loop through all of the timing data, printing it out.
   for (const PrintRecord &Record : llvm::reverse(TimersToPrint)) {
-    Record.Time.print(Total, OS);
-    OS << Record.Description << '\n';
+    if (const TimeRecord &TR = Record.Time;
+        TR.getUserTime() > minPrintTime() ||
```

Comment:
could we call `minPrintTime` only once here (i.e. factoring out this function call)? It's unlikely that compiler would CSE these calls since `minPrintTime` is accessing a ManagedStatic.